test/functional/plugin/lsp_spec.lua

```diff
@@ -987,46 +987,147 @@ describe('LSP', function()
     end)
   end)
   describe('completion_list_to_complete_items', function()
+    local function create_compl_item(opt)
+        return {
+          abbr = opt.abbr,
+          dup = opt.dup or 1,
+          empty = opt.empty or 1,
+          icase = opt.icase or 1,
+          info = opt.info or ' ',
+          kind = opt.kind or 'Unknown',
+          menu = opt.menu or '',
+          word = opt.word,
+          user_data = {
+            nvim = {
+              lsp = {
+                completion_item = {
+                  label = opt.label,
+                  insertText = opt.insertText,
+                  insertTextFormat = opt.insertTextFormat,
+                  textEdit = opt.textEdit,
+                }
+              }
+            }
+          },
+        }
+    end
+    local function get_completion_list(completion_list, prefix)
+      return exec_lua([[return vim.lsp.util.text_document_completion_list_to_complete_items(...)]], completion_list, prefix)
+    end
     -- Completion option precedence:
     -- textEdit.newText > insertText > label
     -- https://microsoft.github.io/language-server-protocol/specifications/specification-current/#textDocument_completion
-    it('should choose right completion option', function ()
+    it('should choose right completion option for label', function ()
       local prefix = 'foo'
       local completion_list = {
         -- resolves into label
         { label='foobar' },
         { label='foobar', textEdit={} },
+      }
+      local completion_list_items = {items=completion_list}
+      local expected = {
+        create_compl_item { abbr = 'foobar', word = 'foobar', label = 'foobar' },
+        create_compl_item { abbr = 'foobar', word = 'foobar', label = 'foobar', textEdit = {} },
+      }
+
+      eq(expected, get_completion_list(completion_list, prefix))
+      eq(expected, get_completion_list(completion_list_items, prefix))
+      eq({}, get_completion_list({}, prefix))
+    end)
+    it('should choose right completion option for insertText', function ()
+      local prefix = 'foo'
+      local completion_list = {
         -- resolves into insertText
         { label='foocar', insertText='foobar' },
         { label='foocar', insertText='foobar', textEdit={} },
+      }
+      local completion_list_items = {items=completion_list}
+      local expected = {
+        create_compl_item { abbr = 'foocar', word = 'foobar', label='foocar', insertText = 'foobar' },
+        create_compl_item { abbr = 'foocar', word = 'foobar', label='foocar', insertText='foobar', textEdit={} },
+      }
+
+      eq(expected, get_completion_list(completion_list, prefix))
+      eq(expected, get_completion_list(completion_list_items, prefix))
```

Comment:
I don't love that every test has this same idea here, as well as the `completion_list_items = {...}` stuff. is there any way to make this more clear what is happening?